src/main/java/teammates/ui/webapi/WebApiServlet.java

```diff
@@ -105,7 +105,7 @@ private void invokeServlet(HttpServletRequest req, HttpServletResponse resp) thr
             log.warning(enfe.getClass().getSimpleName() + " caught by WebApiServlet: "
                     + TeammatesException.toStringWithStackTrace(enfe));
             throwError(resp, HttpStatus.SC_NOT_FOUND, enfe.getMessage());
-        } catch (DeadlineExceededException | DatastoreTimeoutException e) {
+        } catch (DeadlineExceededException | DatastoreException e) {
```

Comment:
This block translates the exception to `504 Gateway Timeout` error, but I believe only a small subset of `DatastoreException` fits into that.

Reply:
Based on this [file](https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto) only `DEADLINE_EXCEEDED` maps to `504 Gateway Timeout`.  I'm thinking we can handle `DatastoreException` separately in a new `catch` clause and generate specific error message depending on the type of datastore error.